lib/iris/fileformats/pp.py

```diff
@@ -1040,7 +1040,8 @@ def _data_bytes_to_shaped_array(data_bytes, lbpack, boundary_packing,
 
     # Mask the array?
     if mdi in data:
-        data = array_masked_to_nans(data, data == mdi)
+        data = np.ma.masked_array(data, data == mdi)
```

Comment:
So all we're trying to do here is replace instances of `mdi` values with `nan` values (and cast int --> float). Both ways that have been tried so far have ended up producing some sort of mask that has ultimately been thrown away; I wonder if we need another way of making this conversion?

As a follow-on question; do we need `array_masked_to_nans` at all - will there _ever_ be masked arrays at any of the places this function is used that will need to be nan-filled?